src/utils/format-numbers.js

```diff
@@ -4,38 +4,42 @@
 
 // @flow
 
+import type { Microseconds, Milliseconds } from '../types/units';
+
 /**
  * Format a positive float into a string.
  *
- * Try to format the value to 2 significant digits as much as possible but
- * without using scientific notation.  The number of decimal places depends
- * on the value: the closer to zero the value is, the more decimal places
- * are used in the resulting string.
+ * Try to format the value to num_digits significant digits as much as
+ * possible but without using scientific notation.  The number of decimal
+ * places depends on the value: the closer to zero the value is, the more
+ * decimal places are used in the resulting string.  No more than max_places
+ * decimal places will be used.
  *
- * For example:
+ * For example, using num_digits = 2 (the default):
  *
  * formatNumber(123,   ) = "123"
  * formatNumber(12.3,  ) =  "12"
  * formatNumber(1.23,  ) =   "1.2"
  * formatNumber(0.01234) =   "0.012"
  */
-export function formatNumber(value: number): string {
-  let result: string;
-  if (value >= 10) {
-    result = value.toFixed(0);
-  } else if (value >= 1) {
-    result = value.toFixed(1);
-  } else if (value >= 0.1) {
-    result = value.toFixed(2);
-  } else {
-    result = value.toFixed(3);
+export function formatNumber(
+  value: number,
+  num_digits: number = 2,
+  max_places: number = 3
+): string {
+  const num_digits_on_left = Math.floor(Math.log10(value)) + 1;
+  let places = num_digits - num_digits_on_left;
+  if (places < 0) {
+    places = 0;
+  } else if (places > max_places) {
+    places = max_places;
   }
 
-  return result;
+  return value.toFixed(places);
 }
 
 export function formatPercent(value: number): string {
-  return formatNumber(value * 100) + '%';
+  return (value * 100).toFixed(0) + '%';
```

Comment:
Can you please explain why this change? What happens for small values, isn't they worth having digits ?
Maybe `formatNumber(value * 100, 1, 3)` ?

Reply:
I found that 0.00% looked weird. and while I don't mind 1.2% or 1.23%, 87.65% was a bit excessive.  Which is why we'd use sagnificant digits.  Maybe two sagnificant digits and no more than one fractional digit?  So formatNumber(value * 100, 2, 1) ?

Reply:
Oh, and when you calculate a percentage I figured that even when the result is 1.23% the 0 in the tens and hundreds place actually count as sagnificant digits.  So _if_ it's strictly by sagnificant digits I think that's how it should be.  I didn't write this before because I handn't thought of the words to explain it.

Reply:
I realize that even for other units (eg: "0.000ms") this isn't useful. Maybe we need to return a "< 0.001" in some cases. But not for this patch.